src/utils.jl

```diff
@@ -1,7 +1,52 @@
-const CatStr = CategoricalString{UInt32}
-struct PooledString end
+struct PooledString <: AbstractString
+    ref::UInt64
+    refs::Vector{String}
+end
+
+Base.show(io::IO, x::PooledString) = print(io, '"', escape_string(x.refs[x.ref]), '"')
 export PooledString
 
+Base.get(x::PooledString) = x.refs[x.ref]
```

Comment:
I don't think this method makes sense here, we use it for categorical values because they wrap another value, but `PooledString` is just a string.

But the `String` and/or `convert` methods should be defined (IIRC the fallbacks are slow since they operate character by character).

Reply:
Yeah, I was just kind of blindly copying the CategoricalString code; I'll go thru and clean it up.